src/main/java/org/embulk/output/sftp/utils/TimeoutCloser.java

```diff
@@ -10,8 +10,7 @@
 
 public class TimeoutCloser implements Closeable
 {
-    @VisibleForTesting
-    int timeout = 300; // 5 minutes
+    private static int timeout = 300; // 5 minutes
```

Comment:
> In my understanding,` BufferedOutputStream` of remote file will be closed by `TimeoutClose` before 600 sec.
`timeout` config is used by `DefaultFileSystemManager`, when it resolves a file. So it's different with this `timeout` value. This `timeout` is the maximum time to wait for a `Closable` to be closed (in this case, it's `BufferedOutputStream`, but I also use it for `FileObject`). The reason to timeout closing op is when connection has issue, it will hang forever (during flush phase).

> And how does this work when `appendFile(localTempFile, remoteFile, outputStream)` works normally but exceeded 300 sec?
During `appendFile`, it will keep writing buffer (each 32MB per write): https://github.com/embulk/embulk-output-sftp/blob/443c44c1c8d2631e8aaea85a8f2fdf5f8ab3d326/src/main/java/org/embulk/output/sftp/SftpUtils.java#L207
When one write is timed out (ie. exceeded 5 mins, because 32MB should be written in just a few seconds), `appendFile` will throw `IOException` with cause as `TimeoutException`.

Reply:
Make sense. Thanks!